src/AspNetCoreModuleV2/AspNetCore/src/applicationmanager.cxx

```diff
@@ -251,25 +250,9 @@ APPLICATION_MANAGER::RecycleApplicationFromManager(
         // Removed the applications which are impacted by the configurtion change
         m_pApplicationInfoHash->DeleteIf(FindConfigChangedApplication, (PVOID)&context);
 
-        if (dwPreviousCounter != m_pApplicationInfoHash->Count())
```

Comment:
I'm still a bit confused on why you removed this. Can you clarify?

Reply:
in the past, we did not create an applicationinfo instance and add it to the hash table in case of hostingmodel conflict and some other cases, i.e., we did not cache the failure state.  This leads to an issue that we tries to create applicationinfo again and again for such failure case.  If there is app recycle happened, we have to check whether the recycled app is in out hashtable or not.  This is why we did not counter check.
In the new code, we always create an applicationinfo instance and insert it into the hash table, no matter the application can be created or not so that we can cache the failure state.  We thus do not need to check the hashtable counter.